cmd/kubectl_token.go

```diff
@@ -579,56 +627,73 @@ func generateKey() (string, error) {
 	return string(token), nil
 }
 
-func getTLSConfig(input *LoginInput) (*tls.Config, error) {
-	config := &tls.Config{}
-	if input.skipVerify || input.caCerts == "" {
-		config = &tls.Config{
-			InsecureSkipVerify: true,
-		}
+// getClient return a client with the provided TLS configuration
+func getClient(skipVerify bool, caCerts string) (*http.Client, error) {
+	tlsConfig, err := getTLSConfig(skipVerify, caCerts)
+	if err != nil {
+		return nil, err
+	}
+
+	// clone the DefaultTransport to get the default values
+	transport := http.DefaultTransport.(*http.Transport).Clone()
+	transport.TLSClientConfig = tlsConfig
+	return &http.Client{Transport: transport}, nil
+}
+
+func getTLSConfig(skipVerify bool, caCerts string) (*tls.Config, error) {
+	config := &tls.Config{
+		InsecureSkipVerify: skipVerify,
+	}
+
+	if caCerts == "" {
 		return config, nil
 	}
 
-	if input.caCerts != "" {
-		cert, err := loadAndVerifyCert(input.caCerts)
-		if err != nil {
-			return nil, err
-		}
-		roots := x509.NewCertPool()
-		ok := roots.AppendCertsFromPEM([]byte(cert))
-		if !ok {
-			return nil, err
-		}
-		config.RootCAs = roots
+	// load custom certs
+	cert, err := loadAndVerifyCert(caCerts)
+	if err != nil {
+		return nil, err
+	}
+
+	// enrichman: should we load the system CertPool instead of creating a new one?
+	roots := x509.NewCertPool()
+	ok := roots.AppendCertsFromPEM([]byte(cert))
```

Comment:
Epinio tried to get and extend the system cert pool first, and then fell back to extending a new empty cert pool

Reply:
I agree. I would like a confirmation from @pjbgf because I'm not sure if I misinterpreted a comment from him.